src/Model/Payload.php

```diff
@@ -113,4 +110,39 @@ public function getCustomData(): array
     {
         return $this->customData;
     }
+
+    /**
+     * Guard the custom data upon instantiation.
+     *
+     * @param array $data
+     */
+    private function guardCustomData(array $data)
+    {
+        if (false === empty($data)) {
+            foreach ($data as $key => $value) {
+                $this->guardValue($value);
+                $this->guardKey($key);
+            }
+        }
+    }
+
+    private function guardValue($value)
```

Comment:
Please rename to `validateCustomDataValue` and add doc comment and return typehint (void) for this method.